sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py

```diff
@@ -288,8 +287,6 @@ def create_database(  # pylint:disable=docstring-missing-param
 
         _set_throughput_options(offer=offer_throughput, request_options=request_options)
         result = self.client_connection.CreateDatabase(database=dict(id=id), options=request_options, **kwargs)
-        if response_hook:
-            response_hook(self.client_connection.last_response_headers)
```

Comment:
We may have to special-case the create/delete/get database operations, as these are the only response hooks that don't take the payload as a parameter, just the headers.

Reply:
what do we do then? do we keep them using the last_response_headers that get cached/ are not thread safe? or make new internal methods that return the headers out? this was probably an oversight in the initial design of the SDK as well... 😞

Reply:
I would be ok with leaving the non-thread safe headers only because customers are generally not using database-level operations for anything - most just call the proxy and go from there